packages/xrpl/test/connection.test.ts

```diff
@@ -68,250 +136,366 @@ describe('Connection', function () {
+      if (isBrowser) {
+        return
+      }
+      const server = await createServer()
       const port = (server.address() as net.AddressInfo).port
       const options = {
         proxy: `ws://127.0.0.1:${port}`,
         authorization: 'authorization',
         trustedCertificates: ['path/to/pem'],
       }
-      const connection = new Connection(this.client.connection.url, options)
+      const connection = new Connection(
+        // @ts-expect-error -- Testing private member
+        clientContext.client.connection.url,
+        options,
+      )
       const expect = 'CONNECT localhost'
 
-      server.on('connection', (socket) => {
-        socket.on('data', (data) => {
-          const got = data.toString('ascii', 0, expect.length)
-          assert.strictEqual(got, expect)
-          server.close()
-          connection.disconnect()
-          done()
+      const connectionPromise = new Promise<void>((resolve) => {
+        server.on('error', () => {
+          destroyServer(server).then(() => {
+            resolve()
+          })
+        })
+        server.on('connection', (socket) => {
+          socket.on('data', (data) => {
+            const got = data.toString('ascii', 0, expect.length)
+            assert.strictEqual(got, expect)
+            if (connection.isConnected()) {
+              destroyServer(server)
+                .then(async () => {
+                  return connection.disconnect().catch((error) => {
+                    // eslint-disable-next-line no-console -- Test
+                    console.error('Failed to disconnect')
+                    throw error
+                  })
+                })
+                .then(() => {
+                  resolve()
+                })
+            } else {
+              destroyServer(server).then(resolve)
+            }
+          })
         })
       })
 
-      connection.connect().catch((err) => {
+      await connection.connect().catch((err) => {
         assert(err instanceof NotConnectedError)
       })
-    }, done)
-  })
-
-  it('Multiply disconnect calls', async function () {
-    this.client.disconnect()
-    this.client.disconnect()
-  })
 
-  it('reconnect', function () {
-    this.client.connection.reconnect()
-  })
-
-  it('NotConnectedError', async function () {
-    const connection = new Connection('url')
-    return connection
-      .request({
-        command: 'ledger',
-        ledger_index: 'validated',
-      })
-      .then(() => {
-        assert.fail('Should throw NotConnectedError')
-      })
-      .catch((error) => {
-        assert(error instanceof NotConnectedError)
-      })
-  })
+      await connectionPromise
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'Multiply disconnect calls',
+    async () => {
+      await clientContext.client.disconnect()
+      await clientContext.client.disconnect()
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'reconnect',
+    async () => {
+      await clientContext.client.connection.reconnect()
```

Comment:
Should this require us to ensure we're connected first?

Reply:
That happens in the setup of the client